docs/creating-a-runtime-check.md

```diff
@@ -0,0 +1,181 @@
+
+Both these interfaces are implemented in the `Abstract_Runtime_Check` class, which developers can use when building out their own runtime checks. This class defines both the `prepare()` and `run()` methods which are required to be implemented for every runtime check.
+
+Below is the basic scaffold when creating a custom runtime check.
+
+```php
+use WordPress\Plugin_Check\Checker\Checks\Abstract_Runtime_Check;
+
+class Custom_Check extends Abstract_Runtime_Check {
+  public function prepare() {
+    // Handle the Checks preparations and return a cleanup function.
+  }
+
+  public function run( Check_Results $results ) {
+    // Handle running the check and adding warnings or errors to the results.
+  }
+}
+```
+
+## Preparations
+
+Preparations are used to prepare the environment before running the check. Preparations can include any logic from activating specific themes or plugins, or creating test content required to perform the check.
+
+Preparations will use the `Preparation` interface which defines a single `prepare()` method. This method is required to be implemented in runtime checks and handles any logic needed to prepare the environent.
+
+All preparations return a clean up function that reverts the changes made by the method.
+
+When creating preparations there are 3 approachs that can be used depending on circumstance.
+
+### Check Preperations
+
+Check preparations are created using the `prepare()` method within a check class.
+
+This type of preparation is useful for carrying out the logic to prepare the environment that is specific to the check that implements it.
+
+Below is an example of a preparation that creates a test post and returns a cleanup function that deletes that post after the check is run.
+
+```php
+use WordPress\Plugin_Check\Checker\Checks\Abstract_Runtime_Check;
+
+class Custom_Check extends Abstract_Runtime_Check {
+  /**
+   * Check Preparation.
+   */
+  public function prepare() {
+    // Create a test post ahead of running the check.
+    $post_id = wp_insert_post(
+      array(
+        'post_title'   => 'Test post',
+        'post_content' => 'This is a test post.',
+        'post_type'    => 'post',
+      )
+    );
+
+    // Return the clean up function.
+    return function() use ( $post_id ) {
+      // Remove the test post created.
+      wp_delete_post( $post_id );
+    }
+  }
+}
+```
+
+### Preparation Classes
+
+Preparation classes are individual classes created to handle specific preparation logic in a single place.
+
+The [`Force_Single_Plugin_Preparation`](https://github.com/10up/plugin-check/blob/trunk/includes/Checker/Preparations/Force_Single_Plugin_Preparation.php) in the Plugin Checker is an example of a preparation class.
+
+Preparation classes are useful for implementing the same preparation logic across multiple checks using the concept of shared preparations.
+
+### Shared Preparations
```

Comment:
These aren't really 2 separate ways: Preparation classes are needed to use the concept of shared preparations, so I think these sections need to be combined. Shared preparations require separation classes - you may not always need to implement a new preparation if what you want to do can be achieved with an existing preparation class, but conceptually they belong together.

So overall, there are only two ways to deal with preparations, either a check implements `Preparation` and comes with its own `prepare()` method, or a check implements `With_Shared_Preparations` and uses shared preparations, in which case writing a `Preparation` class may be needed.

Worth noting that both can also be combined, a check could have shared preparations plus a check-specific preparation.